lib/Dialect/FIRRTL/Transforms/IMConstProp.cpp

```diff
@@ -363,6 +501,29 @@ LatticeValue IMConstPropPass::getExtendedLatticeValue(Value value,
   return LatticeValue(IntegerAttr::get(destType.getContext(), resultConstant));
 }
 
+void IMConstPropPass::markSubelementAccessOp(Operation *subelementAccess) {
+  auto result = subelementAccess->getResult(0);
+  auto fieldRef = getFieldRefFromValueWrapper(result);
+  if (!result.getType().cast<FIRRTLType>().isGround()) {
+    // It is not used by other than subaccess or subfield op, it is necessary to
+    // record the dependency.
+    if (llvm::all_of(result.getUsers(), [](Operation *op) {
+          return isa<SubaccessOp, SubfieldOp>(op);
+        }))
+      return;
+
+    foreachFIRRTLGroundType(
+        result.getType().cast<FIRRTLType>(),
+        [&](unsigned relativeFieldID, auto) {
+          fieldRefToSubelementValues[fieldRef.getSubField(relativeFieldID)]
```

Comment:
I prepared this map to avoid unnecessary propagations between subWhatever ops. This implementation doesn't require propagation between subWhatever ops and their parent ops since the states are shared as FieldRef.